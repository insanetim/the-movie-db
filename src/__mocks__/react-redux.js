export const dispatch = jest.fn()

export const useDispatch = jest.fn(() => dispatch)
export const useSelector = jest.fn(fn => fn())
