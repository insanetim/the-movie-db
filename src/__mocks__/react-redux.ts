const dispatch = jest.fn()
const getState = jest.fn()

const useDispatch = jest.fn(() => dispatch)
const useSelector = jest.fn(fn => fn())

export { dispatch, getState, useDispatch, useSelector }
