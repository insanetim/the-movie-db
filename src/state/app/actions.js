import { v4 as uuid } from 'uuid'

import { NOTIFICATION_DURATION, NOTIFICATION_TYPE } from 'src/constants'

import * as types from './types'

export const showModal = payload => ({
  type: types.SHOW_MODAL,
  payload
})

export const hideModal = () => ({
  type: types.HIDE_MODAL
})

export const showNotification = ({
  id = uuid(),
  messageType = NOTIFICATION_TYPE.SUCCESS,
  messageText,
  duration = NOTIFICATION_DURATION
}) => ({
  type: types.SHOW_NOTIFICATION,
  payload: {
    id,
    messageType,
    messageText,
    duration
  }
})

export const hideNotification = payload => ({
  type: types.HIDE_NOTIFICATION,
  payload
})
