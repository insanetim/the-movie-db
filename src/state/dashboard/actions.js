import * as types from './types'

export const fetchTrending = page => ({
  type: types.FETCH_TRENDING,
  payload: page
})

export const fetchTrendingRequest = () => ({
  type: types.FETCH_TRENDING_REQUEST
})

export const fetchTrendingSuccess = movies => ({
  type: types.FETCH_TRENDING_SUCCESS,
  payload: movies
})

export const fetchTrendingError = error => ({
  type: types.FETCH_TRENDING_ERROR,
  payload: error
})

export const setTrendingPage = page => ({
  type: types.SET_TRENDING_PAGE,
  payload: page
})

export const fetchSearch = ({ page, query }) => ({
  type: types.FETCH_SEARCH,
  payload: { page, query }
})

export const fetchSearchRequest = () => ({
  type: types.FETCH_SEARCH_REQUEST
})

export const fetchSearchSuccess = movies => ({
  type: types.FETCH_SEARCH_SUCCESS,
  payload: movies
})

export const fetchSearchError = error => ({
  type: types.FETCH_SEARCH_ERROR,
  payload: error
})

export const setSearchPage = page => ({
  type: types.SET_SEARCH_PAGE,
  payload: page
})
