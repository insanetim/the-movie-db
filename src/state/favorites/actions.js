import * as types from './types'

export const fetchFavorites = page => ({
  type: types.FETCH_FAVORITES,
  payload: page
})

export const fetchFavoritesRequest = page => ({
  type: types.FETCH_FAVORITES_REQUEST,
  payload: page
})

export const fetchFavoritesSuccess = movies => ({
  type: types.FETCH_FAVORITES_SUCCESS,
  payload: movies
})

export const fetchFavoritesFailure = error => ({
  type: types.FETCH_FAVORITES_FAILURE,
  payload: error
})

export const setFavoritesPage = page => ({
  type: types.SET_FAVORITES_PAGE,
  payload: page
})
