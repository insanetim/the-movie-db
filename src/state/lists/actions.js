import * as types from './types'

export const fetchLists = ({ page, callback }) => ({
  type: types.FETCH_LISTS,
  payload: { page, callback }
})

export const fetchListsRequest = page => ({
  type: types.FETCH_LISTS_REQUEST,
  payload: page
})

export const fetchListsSuccess = movies => ({
  type: types.FETCH_LISTS_SUCCESS,
  payload: movies
})

export const fetchListsFailure = error => ({
  type: types.FETCH_LISTS_FAILURE,
  payload: error
})

export const setListsPage = page => ({
  type: types.SET_LISTS_PAGE,
  payload: page
})

export const fetchList = listId => ({
  type: types.FETCH_LIST,
  payload: listId
})

export const fetchListRequest = () => ({
  type: types.FETCH_LIST_REQUEST
})

export const fetchListSuccess = movies => ({
  type: types.FETCH_LIST_SUCCESS,
  payload: movies
})

export const fetchListFailure = error => ({
  type: types.FETCH_LIST_FAILURE,
  payload: error
})

export const createList = ({ value, callback }) => ({
  type: types.CREATE_LIST,
  payload: { value, callback }
})

export const addToList = ({ listId, movieId }) => ({
  type: types.ADD_TO_LIST,
  payload: { listId, movieId }
})

export const removeFromList = ({ listId, movieId }) => ({
  type: types.REMOVE_FROM_LIST,
  payload: { listId, movieId }
})

export const deleteList = ({ listId, callback }) => ({
  type: types.DELETE_LIST,
  payload: { listId, callback }
})
