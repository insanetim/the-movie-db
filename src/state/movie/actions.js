import * as types from './types'

export const fetchMovie = movieId => ({
  type: types.FETCH_MOVIE,
  payload: movieId
})

export const fetchMovieRequest = () => ({
  type: types.FETCH_MOVIE_REQUEST
})

export const fetchMovieSuccess = movie => ({
  type: types.FETCH_MOVIE_SUCCESS,
  payload: movie
})

export const fetchMovieFailure = error => ({
  type: types.FETCH_MOVIE_FAILURE,
  payload: error
})

export const fetchMovieStates = movieId => ({
  type: types.FETCH_MOVIE_STATES,
  payload: movieId
})

export const updateMovieStates = movieStates => ({
  type: types.UPDATE_MOVIE_STATES,
  payload: movieStates
})

export const changeMovieInFavorites = ({ movieId, inFavorites }) => ({
  type: types.CHANGE_MOVIE_IN_FAVORITES,
  payload: { movieId, inFavorites }
})

export const changeMovieInWatchlist = ({ movieId, inWatchlist }) => ({
  type: types.CHANGE_MOVIE_IN_WATCHLIST,
  payload: { movieId, inWatchlist }
})
