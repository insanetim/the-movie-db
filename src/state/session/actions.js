import { createAsyncThunk } from '@reduxjs/toolkit'
import Cookies from 'js-cookie'
import { or, path } from 'ramda'

import httpClient from 'src/api/httpClient'
import * as endpoints from 'src/constants/endpoints'
import { showNotification } from 'src/state/app/actions'
import { sessionIdSelector } from './selectors'
import * as types from './types'

export const setSession = sessionId => ({
  type: types.SET_SESSION,
  payload: sessionId
})

export const deleteSession = () => ({
  type: types.DELETE_SESSION
})

export const fetchAccount = () => ({
  type: types.FETCH_ACCOUNT
})

export const fetchAccountSuccess = account => ({
  type: types.FETCH_ACCOUNT_SUCCESS,
  payload: account
})

export const loadingOn = () => ({
  type: types.LOADING_ON
})

export const loadingOff = () => ({
  type: types.LOADING_OFF
})

export const logIn = createAsyncThunk(types.LOG_IN, async (userData, { dispatch }) => {
  dispatch(loadingOn())

  try {
    const { data: requestToken } = await httpClient.get(endpoints.createRequestToken)
    const { data: sessionToken } = await httpClient.post(endpoints.createSessionWithLogin, {
      request_token: requestToken.request_token,
      ...userData
    })
    const { data } = await httpClient.post(endpoints.createSession, { request_token: sessionToken.request_token })
    Cookies.set('session_id', data.session_id)
    dispatch(setSession(data.session_id))
  } catch (error) {
    const errorMessage = or(path(['response', 'data', 'status_message'], error), error.message)
    dispatch(showNotification({ messageType: 'error', messageText: errorMessage }))
  }

  dispatch(loadingOff())
})

export const logOut = createAsyncThunk(types.LOG_OUT, async (_, { dispatch, getState }) => {
  const sessionId = sessionIdSelector(getState())

  try {
    await httpClient.delete(endpoints.deleteSession, { data: { session_id: sessionId } })
    Cookies.remove('session_id')
    dispatch(deleteSession())
  } catch (error) {
    const errorMessage = or(path(['response', 'data', 'status_message'], error), error.message)
    dispatch(showNotification({ messageType: 'error', messageText: errorMessage }))
  }
})
