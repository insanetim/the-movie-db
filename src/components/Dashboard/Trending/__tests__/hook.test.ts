import { act, renderHook } from '@testing-library/react'
import { useSearchParams } from 'react-router-dom'
import { dispatch } from 'src/__mocks__/react-redux'
import * as dashboardActions from 'src/store/dashboard/actions'

import useContainer from '../hook'

jest.mock('src/store/dashboard/selectors', () => ({
  dashboardMoviesSelector: () => null
}))

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useSearchParams: jest.fn()
}))
const searchParams = new URLSearchParams()
const setSearchParams = jest.fn()
jest.mocked(useSearchParams).mockReturnValue([searchParams, setSearchParams])

describe('Trending useContainer hook', () => {
  it('should match snapshot', () => {
    const { result } = renderHook(useContainer)

    expect(result.current).toMatchSnapshot()
  })

  it('should check `handlePagination` method', () => {
    const { result } = renderHook(useContainer)

    act(() => {
      result.current.handlePagination(3)
    })

    expect(setSearchParams).toHaveBeenCalledWith({ page: '3' })
  })

  it('should check `useEffect` method', () => {
    const fetchTrending = jest.spyOn(dashboardActions, 'fetchTrending')
    renderHook(useContainer)

    expect(dispatch).toHaveBeenCalled()
    expect(fetchTrending).toHaveBeenCalledWith('1')
  })
})
