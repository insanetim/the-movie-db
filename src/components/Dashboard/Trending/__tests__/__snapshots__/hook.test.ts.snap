// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trending useContainer hook should match snapshot 1`] = `
{
  "error": null,
  "handlePagination": [Function],
  "loading": true,
  "movies": null,
}
`;
